server/apps/entry/serializers.py

```diff
@@ -120,6 +120,45 @@ class Meta:
         model = Figure
         fields = '__all__'
 
+    def validate_figure_country(self, attrs):
+        errors = OrderedDict()
+        country = attrs.get('country')
+        if not country:
+            return errors
+        entry = attrs.get('entry', getattr(self.instance, 'entry', None))
+        if country not in entry.event.countries.all():
+            errors.update(
+                {
+                    'country': 'Choose country within event'
+                }
+            )
+        return errors
+
+    def validate_figure_geo_locations(self, attrs):
+        errors = OrderedDict()
+        if not attrs.get('geo_locations') or not attrs.get('country'):
+            return errors
```

Comment:
Same here...
get from `attrs`, if not in attrs, get from db
```suggestion
        geo_locations = attrs.get('geo_locations')
        if not geo_locations and self.instance:
            geo_locations = self.instance.geo_locations.all()
```

Reply:
Same for country